src/analysis/souffle/operations.dl

```diff
@@ -27,28 +28,42 @@
 // regardless of arity. Useful for rules that apply to any operation.
 .type OperandList = [ operand: AccessPath, next: OperandList ]
 
-// Operations of any arity go in this relation.
-.decl operation(owner: Principal, operator: Operator, result: AccessPath, operandList: OperandList)
+// An Operation.
+.type Operation = [owner: Principal, operator: Operator, result: AccessPath,
+                   operandList: OperandList, attributes: AttributeList]
+
+// Operation universe.
+.decl isOperation(operation: Operation)
```

Comment:
Is this essentially a new way to declare an `operation` relation?

Reply:
Yes, this is a direct replacement of the `operation` relation. I'm just changing the name to make it more clear that this is the universe of all `Operation`s. I expect that we will eventually want to make this an input parameter and the IR translator will populate this universe for us.